src/core.jl

```diff
@@ -123,9 +141,13 @@ function Base.run(x::JSONRPCEndpoint)
             Base.display_error(stderr, err, bt)
         end
     end
+
+    x.status = :running
 end
 
 function send_notification(x::JSONRPCEndpoint, method::AbstractString, params)
+    x.status == :running || error("Endpoint is not running.")
```

Comment:
Define and use `isrunning(x::JSONRPCEndpoint)` instead ? It would be more robust for the typo in symbol in general.